packages/selfcare-onboarding-consumer/src/services/selfcareOnboardingProcessor.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  getInteropHeaders,
+  logger,
+  RefreshableInteropToken,
+} from "pagopa-interop-commons";
+import { EachMessagePayload } from "kafkajs";
+import { v4 as uuidv4 } from "uuid";
+import { tenantApi } from "pagopa-interop-api-clients";
+import { genericInternalError } from "pagopa-interop-models";
+import { TenantProcessClient } from "../clients/tenantProcessClient.js";
+import { InstitutionEventPayload } from "../model/institutionEvent.js";
+import { ORIGIN_IPA } from "../model/constants.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function selfcareOnboardingProcessorBuilder(
+  refreshableToken: RefreshableInteropToken,
+  tenantProcessClient: TenantProcessClient,
+  productName: string,
+  allowedOrigins: string[]
+) {
+  return {
+    async processMessage({
+      message,
+      partition,
+    }: EachMessagePayload): Promise<void> {
+      const correlationId = uuidv4();
+
+      const loggerInstance = logger({
+        serviceName: "selfcare-onboarding-consumer",
+        correlationId,
+      });
+
+      try {
+        loggerInstance.info(
+          `Consuming message for partition ${partition} with offset ${message.offset}`
+        );
+
+        if (!message.value) {
+          loggerInstance.warn(
+            `Empty message for partition ${partition} with offset ${message.offset}`
+          );
+          return;
+        }
+
+        const stringPayload = message.value.toString();
+        const jsonPayload = JSON.parse(stringPayload);
+
+        // Process only messages of our product
+        // Note: doing this before parsing to avoid errors on messages of other products
+        if (jsonPayload.product !== productName) {
+          loggerInstance.info(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Not required product: ${jsonPayload.product}`
+          );
+          return;
+        }
+
+        const eventPayload = InstitutionEventPayload.parse(jsonPayload);
+
+        const institution = eventPayload.institution;
+        if (allowedOrigins.indexOf(institution.origin) < 0) {
```

Comment:
```suggestion
        if (!allowedOrigins.includes(institution.origin)) {
```
We could simplify this check like this IMHO

Reply:
Yep! I copy-pasted this logic from the other repo but it's worth improving it 🫶